File: nodecore-miners-pop/src/main/java/nodecore/miners/pop/contracts/BlockStore.java
// VeriBlock NodeCore
// Copyright 2017-2019 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package nodecore.miners.pop.contracts;

import java.util.concurrent.atomic.AtomicReference;

public class BlockStore {
    private AtomicReference<VeriBlockHeader> chainHead = new AtomicReference<>();
    public VeriBlockHeader getChainHead() {
        return chainHead.get();
    }
    public void setChainHead(VeriBlockHeader blockBlockHeader) {
        chainHead.set(blockBlockHeader);
    }

    public BlockStore() {

    }
}

File: nodecore-miners-pop/src/main/java/nodecore/miners/pop/contracts/BlockStore.kt
// VeriBlock NodeCore
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
package nodecore.miners.pop.contracts

import java.util.concurrent.atomic.AtomicReference

class BlockStore {
    private val chainHead = AtomicReference<VeriBlockHeader>()

    fun getChainHead(): VeriBlockHeader? {
        return chainHead.get()
    }

    fun setChainHead(blockBlockHeader: VeriBlockHeader) {
        chainHead.set(blockBlockHeader)
    }
}
